### DISCUSSION TITLE
Log"系统版本"貌似有问题（？

### DISCUSSION THREAD

**Discussion by Ad-closeNN** [Meloong-Git/PCL#4332] [提问]
`[17:32:37.517] [System] DPI：96，系统版本：10.0.22631.0，PCL 位置：D:\众 多 m c 版 本\1.12.1\Plain Craft Launcher 2.exe`

`系统版本：10.0.22631.0`在哪看 总觉得怪怪的（因为是22631.0，可是winver 提示是22631.3737）

![image](https://github.com/user-attachments/assets/90c8706f-f5b9-4358-9de0-cbefc9935414)

~~本来还没想着要发出去的结果不小心敲到了Enter还删不掉~~

**Comment by 3gf8jv4dv** [comment]
https://github.com/Hex-Dragon/PCL2/blob/5bd8ac3870170b0acd4d81934227829f8e79ca84/Plain%20Craft%20Launcher%202/Modules/Base/ModBase.vb#L85
https://github.com/Hex-Dragon/PCL2/blob/5bd8ac3870170b0acd4d81934227829f8e79ca84/Plain%20Craft%20Launcher%202/FormMain.xaml.vb#L427

---

- [Environment.OSVersion Property (System) | Microsoft Learn](https://learn.microsoft.com/en-us/dotnet/api/system.environment.osversion?view=netframework-4.8.1)
- [Breaking change: Environment.OSVersion returns the correct operating system version - .NET | Microsoft Learn](https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/5.0/environment-osversion-returns-correct-version)

Looks like expected behavior for Visual Basic .NET. The example shown in the second link is the same as your situation.

*If I am wrong, please point it out.*

**Comment by Ad-closeNN** [comment]
虽然看不懂，但大受震撼（不知道也不想懂所以close了吧

**Reply by MoYuan-CN** [reply]
`Environment.OSVersion.Version` 返回 `{Major}.{Minor}.{Build}.{Revision}`，不清楚为什么 `Revision` 是 `0`

**Reply by 3gf8jv4dv** [reply]
Simply put, PCL uses `Environment.OSVersion` to determine the current system version. The last digit (revision) of the system version returned in this way is `0`.

The second link references an example showing two version numbers:

- 6.2.9200.0
- 10.0.19042.0

It is the same as the version number you provided.
